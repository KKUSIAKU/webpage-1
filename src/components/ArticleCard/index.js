import React from "react";
import PropTypes from "prop-types";

import style from "./style.scss";


const defaultSrc = "../../public/images/defaultArticleImage.png";
const defaultAlt = "poster image not available";
const ArticleCard = ({
  articleId, title, year, src, alt }) => {
  if (src && !alt) {
    throw new Error("In <Article/> : alt property is not set while src property is found");
  }
  return (
    <article id={articleId}>
      <div className="card-img-container">
        <img
          className="img-fluid"
          src={src ? src : defaultSrc}
          alt={alt ? alt : defaultAlt} />
      </div>
      <header><h3 className="article-title py-2">{title}</h3></header>
      <footer className="d-flex justify-content-around">
        <p>{year}</p>
        <p>{year}</p>
      </footer>
    </article>
  );
};

ArticleCard.propTypes = {
  articleId: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired
};

export default ArticleCard; 
