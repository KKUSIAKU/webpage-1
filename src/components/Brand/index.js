import React from "react";

const Brand = () => (
  <div className="brand-container">
    <a href="#" className="navbar-brand">Brand Name</a>
  </div>
);

export default Brand;
