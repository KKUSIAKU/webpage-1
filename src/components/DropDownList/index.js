import React from "react";

const DropDownList = ({ label,filter, list }) => {
  return (
    <div className="d-flex flex-row">
      <div className="filter-label mx-2">{label}</div>
      <div className="dropdown">
        <button
          className="btn dropdown-toggle"
          type="button" id="dropdownMenuButton"
          data-toggle="dropdown"
          aria-haspopup="true"
          aria-expanded="false">
          {filter}
        </button>
        <ul className="dropdown-menu" aria-labelledby="dropdownMenuButton">
          {list.map((item, index) => <li className="dropdown-item"
            key={`${label}-item-${index}`}>
            {item}
          </li>)}
        </ul>
      </div>
    </div>
  );
};



export default DropDownList;