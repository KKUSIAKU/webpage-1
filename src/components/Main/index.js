import React from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import * as  actions from "../../state/actions.js";
import ArticleCard from "../ArticleCard";
import PageButton from "../PageButton";
import style from "./style.scss";

const url = "/videos";

class Main extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.fetchAllMovies.bind(null, url,{page:0})();
  }

  render() {
    let movies = this.props.movies[this.props.page];

    return (
      <main>
        <p className="text-info px-5">{`${movies.length} results`} </p>
        <div className="d-flex flex-wrap justify-content-center">
          {movies.map((movie, index) =>
            <div className="articleItem card border-secondary m-1 p-2" key={index}>
              <ArticleCard {...movie} />
            </div>)
          }
        </div>
        <div className="search-pages">
          <PageButton page={1}/>
          <PageButton page={2}/>
          <PageButton page={3}/>
        </div>
      </main>
    );
  }
}


const mapStateToProps = (state) => {
  let {movies, page} = state; 
  return {
    page,
    movies,
  };
};

const mapDispathToProps = (dispatch) => {
  return bindActionCreators(actions, dispatch);
};

export default connect(mapStateToProps, mapDispathToProps)(Main); 