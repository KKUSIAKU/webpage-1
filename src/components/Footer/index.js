import React from "react";
import Brand from "../Brand";
import NavigationBar from "../Navigation"; 

const Footer = () => (
  <footer className="page-footer">
    <div id="page-footer-container">
      <div className="container-fluid">
        <div className="row row-border justify-content-between">
          <div className="brand-container col-2 menu-border">
            <Brand />
          </div>
          <div className="menu-container col-6 menu-border">
            <NavigationBar links = {[
              {href:"#", label:"About Us"}, 
              {href:"#courses", label:"Terms of use"},
              {href:"#mycourses", label:"Policy"},  
              {href:"#sign-in", label:"Contact Us"}
            ]}/>
          </div>
        </div>
      </div>
    </div>
  </footer>);

export default Footer; 