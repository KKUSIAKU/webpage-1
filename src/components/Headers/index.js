import React from "react";
import Brand from "../Brand";
import NavigationBar from "../Navigation"; 

const Header = () => (
  <header className="page-header navbar-light bg-light fixed-top">
    <div id="page-header-container">
      <div className="container-fluid">
        <div className="row row-border justify-content-between">
          <div className="brand-container col-2 menu-border">
            <Brand />
          </div>
          <div className="menu-container col-6 menu-border">
            <NavigationBar links = {[
              {href:"#", label:"HOME"}, 
              {href:"#courses", label:"CATALOG"},
              {href:"#mycourses", label:"MY PURCHASES"},  
              {href:"#sign-in", label:"LOG IN"}, 
              {href:"#sign-up", label:"SIGN UP"}, 
            ]}/>
          </div>
        </div>
      </div>
    </div>
  </header>);

export default Header;
