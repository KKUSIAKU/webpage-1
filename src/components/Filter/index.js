import React from "react";
import DropDownList from "../DropDownList";

const myList = ["item1", "item2", "item3", "item4"]; 
const myLabel = "Genre"; 



class Filter extends React.Component {
  constructor(props) {
    super(props);
    
    this.state = {
      filter: myList[0],
    };
    
    this.list = myList; 
    this.label = myLabel;

    this.setFilter = this.setFilter.bind(this);
    this.selectFiltersOnClick = this.selectFiltersOnClick.bind(this);
  }

  setFilter(filter) {
    this.setState({
      filter: filter
    });
  }

  selectFiltersOnClick(e) {
    this.setFilter(e.target.innerHTML);
  }

  render() {
    return (
      <div className="filter-container d-flex px-3 justify-content-between">
        <div className="filter d-flex">
          <div id="genre-filter" className="mx-2" onClick={this.selectFiltersOnClick}>
            <DropDownList label={myLabel} filter={this.state.filter} list={myList}/>
          </div>
        </div>
      </div>
    );
  }
}

export default Filter; 
