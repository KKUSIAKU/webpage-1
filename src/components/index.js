import Navigation from "./Navigation"; 
import Header from "./Headers";
import Footer from "./Footer";
export {
  Header, 
  Navigation, 
  Footer
};