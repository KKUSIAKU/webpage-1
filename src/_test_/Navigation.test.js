import React from "react";
import { Navigation } from "../components";
import renderer from "react-test-renderer";
import ReactTestUtils from 'react-dom/test-utils';

test("Navigation testing", function(){
  const navigationComponent = renderer.create(
    <Navigation menuId="test" links={[
      {href:"home",label:"home"}, 
      {href:"about", label:"about"}
    ]}/>
  ); 
  let tree = navigationComponent.toJSON(); 
  expect(tree).toMatchSnapshot();
  
});