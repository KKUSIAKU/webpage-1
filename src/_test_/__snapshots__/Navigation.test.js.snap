// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation testing 1`] = `
<nav
  className="navbar navbar-expand-md navbar-light bg-light menu-border"
  id="menu1"
>
  <p />
  <button
    aria-controls="menu"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#test"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="#test"
  >
    <ul
      className="navbar-nav  container justify-content-end"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#home"
        >
          home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#about"
        >
          about
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
